src/pages/_layouts/FocusedLayout.vue

```diff
@@ -1,8 +1,29 @@
+<script setup lang="ts">
+import { useRoute } from 'vue-router';
+
+const route = useRoute();
+
+function getReturnRoute() {
+  const queryReturnRoute = route.query?.returnRoute as string;
+
+  if (queryReturnRoute) {
+    const queryReturnParams = route.query?.returnParams as string;
+
+    if (queryReturnParams) {
+      return { name: queryReturnRoute, params: JSON.parse(queryReturnParams) };
+    }
+
+    return { name: queryReturnRoute };
+  }
+
+  return { name: 'home' };
```

Comment:
I need this function because the migrate button appears in the homepage / pool page and when you close it you need to navigate the user to the originating page. (Also, can't use something like history.back or history.go(-1) because a user can refresh the page)